pkgs/by-name/ba/baballonia/package.nix

```
@@ -0,0 +1,122 @@
+{
+  cmake,
+  opencv,
+  udev,
+  libjpeg,
+  libGL,
+  fontconfig,
+  xorg,
+  lib,
+  buildDotnetModule,
+  dotnetCorePackages,
+  fetchFromGitHub,
+  fetchurl,
+  stdenv,
+}:
+let
+  internal = fetchurl {
+    # This URL is weird but this is the primary source
+    url = "http://217.154.52.44:7771/builds/trainer/1.0.0.0.zip";
+    sha256 = "sha256:0cfc1r1nwcrkihmi9xn4higybyawy465qa6kpls2bjh9wbl5ys82";
+  };
+
+  dotnet = dotnetCorePackages.dotnet_8;
+
+  opencvsharp = stdenv.mkDerivation rec {
+    pname = "opencvsharp";
+    version = "4.11.0.20250507";
+
+    src = fetchFromGitHub {
+      owner = "shimat";
+      repo = "opencvsharp";
+      tag = version;
+      hash = "sha256-CkG4Kx/AkZqyhtclMfS51a9a9R+hsqBRlM4fry32YJ0=";
+    };
+    nativeBuildInputs = [ cmake ];
+    buildInputs = [ opencv ];
+    sourceRoot = "${src.name}/src";
+
+    cmakeFlags = [ (lib.cmakeFeature "CMAKE_POLICY_VERSION_MINIMUM" "3.5") ];
+  };
+
+  baballoniaPrograms = [
+    cmake
+    opencv
+    udev
+    libjpeg
+    libGL
+    fontconfig
+    xorg.libX11
+    xorg.libSM
+    xorg.libICE
+    opencvsharp
+  ];
+in
+buildDotnetModule (finalAttrs: {
+  version = "1.1.0.8";
+  pname = "baballonia";
+
+  patches = [ ./0001-disable-auto-updating.patch ];
+
+  buildInputs = [
+    cmake
+    opencv
+    udev
+    libjpeg
+    libGL
+    fontconfig
+    xorg.libX11
+    xorg.libSM
+    xorg.libICE
+    opencvsharp
+  ];
+
+  src = fetchFromGitHub {
+    owner = "Project-Babble";
+    repo = "Baballonia";
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-OnLCK/T7b0NsExKEv95a0lM9TccJkI/uLGIe+oz3Rtw=";
+    fetchSubmodules = true;
+  };
+
+  dotnetSdk = dotnet.sdk;
+  nugetDeps = ./deps.json;
+  dotnetRuntime = dotnet.runtime;
+  projectFile = "src/Baballonia.Desktop/Baballonia.Desktop.csproj";
+
+  runtimeDeps = [ udev ];
+
+  makeWrapperArgs = [
+    "--chdir"
+    "${placeholder "out"}/lib/baballonia"
+  ];
+
+  postUnpack = ''
+    ln -s ${internal} $sourceRoot/src/Baballonia.Desktop/_internal.zip
+  '';
+
+  postFixup = ''
+    mkdir -p $out/lib/baballonia/Modules
+    mv $out/bin/Baballonia.Desktop $out/bin/baballonia
+    mv $out/lib/baballonia/Baballonia.VFTCapture.dll $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.VFTCapture.pdb $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.OpenCVCapture.dll $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.OpenCVCapture.pdb $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.IPCameraCapture.dll $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.IPCameraCapture.pdb $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.SerialCameraCapture.dll $out/lib/baballonia/Modules/
+    mv $out/lib/baballonia/Baballonia.SerialCameraCapture.pdb $out/lib/baballonia/Modules/
```

Maybe we would want to use a wildcard here to avoid unexpected failures in case new libraries get added in the future.


```suggestion
    mv $out/lib/baballonia/Baballonia.*.{dll,pdb} $out/lib/baballonia/Modules/
```

---

@Scrumplex I just wanted to say I did try this in the past, but it moved more things that are necessary. It may be causing https://github.com/NixOS/nixpkgs/pull/459868#issuecomment-3538681064 somehow? 

Though, I'm not sure on how else to improve the situation.
Maybe at least using an array of the names of the things that need to be moved, and just iterating over that?